### ISSUE TITLE
scoping issue in @sync 

### ISSUE THREAD

**Issue by amitmurthy** [JuliaLang/julia#5930]
The below scoping behavior seems a little confusing:

```
addprocs(1)

function foo()
    a = 3
    @sync begin 
        @spawn println(a)
    end
end

foo()
```

works as expected, but 

```
@sync begin
    b = 2
    @spawn println(b)
end
```

fails with a `exception on 2: ERROR: b not defined` . 

Was triggered by https://groups.google.com/d/msg/julia-users/RsX64KwJUBU/iZAwcNB_c5kJ

Maybe related: https://github.com/JuliaLang/julia/issues/3713

**Comment by amitmurthy** [comment]
Closed by #19594

**Comment by tkelman** [comment]
tested?
